RequestService/RequestService.Handlers/PostRequestForHelpHandler.cs

```diff
@@ -0,0 +1,369 @@
+using HelpMyStreet.Utils.Extensions;
+using RequestService.Handlers.BusinessLogic;
+using System.Text;
+using HelpMyStreet.Contracts.GroupService.Request;
+using HelpMyStreet.Contracts.GroupService.Response;
+
+namespace RequestService.Handlers
+{
+    public class PostRequestForHelpHandler : IRequestHandler<PostRequestForHelpRequest, PostRequestForHelpResponse>
+    {
+        private readonly IRepository _repository;
+        private readonly ICommunicationService _communicationService;
+        private readonly IUserService _userService;
+        private readonly IAddressService _addressService;
+        private readonly IGroupService _groupService;
+        private readonly IOptionsSnapshot<ApplicationConfig> _applicationConfig;
+        private readonly IMultiJobs _multiJobs;
+        public PostRequestForHelpHandler(
+            IRepository repository,
+            IUserService userService,
+            IAddressService addressService,
+            ICommunicationService communicationService,
+            IGroupService groupService,
+            IOptionsSnapshot<ApplicationConfig> applicationConfig,
+            IMultiJobs multiJobs)
+        {
+            _repository = repository;
+            _userService = userService;
+            _addressService = addressService;
+            _communicationService = communicationService;
+            _groupService = groupService;
+            _applicationConfig = applicationConfig;
+            _multiJobs = multiJobs;
+        }
+
+        private void AddMultiAndRepeats(PostRequestForHelpRequest request)
+        {
+            var firstHelpRequestDetail = request.HelpRequestDetails.First();
+            var firstJob = firstHelpRequestDetail.NewJobsRequest.Jobs.First();
+            var requestType = firstJob.SupportActivity.RequestType();
+
+            if (requestType == RequestType.Shift &&
+                firstJob.NumberOfRepeats > 1 &&
+                firstJob.RepeatFrequency != Frequency.Once)
+            {
+                _multiJobs.AddShiftRepeats(request.HelpRequestDetails, firstJob.NumberOfRepeats);
+            }
+            else
+            {
+                _multiJobs.AddMultiVolunteers(firstHelpRequestDetail.NewJobsRequest);
+                _multiJobs.AddRepeats(firstHelpRequestDetail.NewJobsRequest, DateTime.UtcNow);
+            }
+
+            foreach (HelpRequestDetail helpRequestDetail in request.HelpRequestDetails)
+            {
+                foreach (Job j in helpRequestDetail.NewJobsRequest.Jobs)
+                {
+                    //add a guid for each job
+                    if (j.Guid == Guid.Empty)
+                    {
+                        j.Guid = Guid.NewGuid();
+                    }
+                };
+            }
+        }
+
+        private void HandleASAP(PostRequestForHelpRequest request)
+        {
+            foreach (Job j in request.HelpRequestDetails.First().NewJobsRequest.Jobs.Where(x => x.DueDateType == DueDateType.ASAP))
```

Comment:
This code seems to be duplicated, here and in AddRepeats, and in AddShiftRepeats.